Sources/Risk/Risk.swift

```diff
@@ -41,22 +45,41 @@ public final class Risk {
         }
     }
     
-    public func publishData (cardToken: String? = nil, completion: @escaping (Result<PublishRiskData, RiskError.Publish>) -> Void) {
-        guard let fingerprintService = fingerprintService else {
-            completion(.failure(.fingerprintServiceIsNotConfigured))
-            return
+    public func publishData(cardToken: String? = nil, completion: @escaping (Result<PublishRiskData, RiskError.Publish>) -> Void) {
+        guard let _ = fingerprintService else {
+          completion(.failure(.fingerprintServiceIsNotConfigured))
+          return
         }
-        
-        fingerprintService.publishData { [weak self] fpResult in
-            guard let self = self else { return }
-            
+
+        DispatchQueue.main.async {
+            // Timer setup remains on the main queue
+            self.timer = Timer.scheduledTimer(withTimeInterval: self.fingerprintTimeoutInterval, repeats: false) { _ in // 2.00
+              
+                self.loggerService.log(riskEvent: .publishFailure, blockTime: self.blockTime, deviceDataPersistTime: nil, fpLoadTime: self.fingerprintService?.fpLoadTime, fpPublishTime: nil, deviceSessionId: nil, requestId: nil, error: RiskLogError(reason: "publishData", message: RiskError.Publish.fingerprintTimeout.localizedDescription, status: nil, type: "Timeout"))
+                completion(.failure(.fingerprintTimeout))
+          }
+        }
+
+        fingerprintService?.publishData { [weak self] fpResult in
+          guard let self = self else { return }
+
+          DispatchQueue.main.async {
+            guard let timer = self.timer, timer.isValid else { // 2.59 -> valid
```

Comment:
Timer would become invalid at `2.59` seconds. We can remove these comments hence they are the things that I forgot to delete :)